File: 0119-pascals-triangle-ii/0119-pascals-triangle-ii.py
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        result = [1]
        before = 1

        for i in range(1, rowIndex + 1):
            num = before * (rowIndex - i + 1) // i
            result.append(num)
            before = num

        return result

File: 0119-pascals-triangle-ii/0119-pascals-triangle-ii.cpp
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<int> result(1, 1);

        for (int i = 1; i < rowIndex + 1; i++) {
            for (int j = i - 1; j > 0; j--) {
                result[j] = result[j] + result[j - 1];
            }
            result.push_back(1);
        }

        return result;
    }
};